m3u8/model.py

```diff
@@ -368,12 +368,8 @@ def dump(self, filename):
 
     def _create_sub_directories(self, filename):
         basename = os.path.dirname(filename)
-        try:
-            if basename:
-                os.makedirs(basename)
-        except OSError as error:
-            if error.errno != errno.EEXIST:
```

Comment:
[This block](https://github.com/python/cpython/blob/a69256527f93d2aa32e76658deab829e324d97b6/Lib/os.py#L226-L230) in the CPython source explains why this check isn't enough:

> Cannot rely on checking for EEXIST, since the operating system could give priority to other errors like EACCES or EROFS

In #291, `EACCES` is the error code rather than `EEXIST`.

Reply:
Good to know! Thank you!